### PR TITLE
Refactor the play loop

### PR DESCRIPTION
**Pull request by drewish**
Here's a series of commits to try to extract some of the logic from the play loop into methods.

### CODE REVIEW

**Comment by MikeTaylor** [comment]
Anyway, I read the individual commits and they all make sense, so I am merging. Thanks for this.

**Comment by MikeTaylor** [comment]
Uh-oh -- now that I accepted this, `rake test` no longer passes.

Can you look into this, please? It seems that occurrences are firing at a different time.

**Comment by drewish** [comment]
Yeah had left separate commits to make the changes easier to follow. I should have called out that the prepare game method had a change to the order of the look vs occurance handling. I wanted to run the change in order by you first. 

Give the test diff a look and see what you think. Happy to revert that commit or fix the order.

**Comment by MikeTaylor** [comment]
I did look at the diff, yes. I'd appreciate it if you'd fix the order.

(And yes, your individual commits were a much easier read than the total diff -- the opposite of what is usually the case in my experience.)

File: lib/scottkit/play.rb [hunk 1]
```diff
@@ -6,27 +6,32 @@ def play
...
-          actually_look
-        end
+
+        actually_look if @need_to_look
 
         return @finished if finished?
 
         print "Tell me what to do ? "
         if !(line = gets)
+          # TODO: Find out when we would we end up here?
```
Comment (drewish) [inline lib/scottkit/play.rb]: Any thoughts on this bit right here? I couldn't figure out how to trigger it.

File: lib/scottkit/play.rb [hunk 2]
```diff
@@ -1,9 +1,43 @@
 module ScottKit
   class Game
-    private
 
     # Returns 1 if the game was won, 0 otherwise
     def play
+      prepare_to_play
+
+      while true
+        # TODO: Find out if this can move to the bottom of the loop. If so we
```
[the remark this answers is not in the document]
Comment (drewish) [reply lib/scottkit/play.rb]: What do you think about having it twice then? Once in `prepare_to_play` for the first frame and then at the bottom of the loop after handling the lighting? I was planning on putting the initial look into `prepare_to_play` to fix the ordering.

Effectively:
```rb
actually_look
run_matching_actions(0, 0)

while !finished?
  actually_look if @need_to_look  

  # grab input
  # process input

  execute_command(words[0], words[1])
  process_lighting
  run_matching_actions(0, 0)
end
@finished

File: lib/scottkit/play.rb (hunk 1 above)
Comment (MikeTaylor) [reply lib/scottkit/play.rb]: `gets` returns `nil` at end of file, so this line handles when a user quits the game by typing CTRL-D (or whatever their terminal software uses to signal EOF), and of course the equally important case when you're piping a sequence of commands into `scottkit`.

So I've replaced your TODO with a comment to that effect.

**Comment by MikeTaylor** [comment]
Meanwhile I prefer to leave the look as `while true` and avoid duplicating the call.

**Pull request merged**